store/src/main/java/org/apache/rocketmq/store/DefaultMessageStore.java

```diff
@@ -163,11 +163,13 @@ public class DefaultMessageStore implements MessageStore {
     private volatile boolean shutdown = true;
     protected boolean notifyMessageArriveInBatch = false;
 
-    private StoreCheckpoint storeCheckpoint;
+    protected StoreCheckpoint storeCheckpoint;
     private TimerMessageStore timerMessageStore;
 
     private final LinkedList<CommitLogDispatcher> dispatcherList;
 
+    private RocksDBMessageStore rocksDBMessageStore;
+
```

Comment:
Creating a new class, such as RocksDBConsumeQueueDoubleWriteMessageStore, and then inheriting from DefaultMessageStore.java, and overriding some methods instead of modifying the original DefaultMessageStore, would be better?

Reply:
把双写理解为DefaultMessageStore的一个小功能，而不是一个独立的MessageStore，这样更简单一点